src/intersection-observer-polyfill.js

```diff
@@ -227,11 +234,22 @@ export class IntersectionObserverPolyfill {
       }
     }
 
-    // push new observed element
-    this.observeEntries_.push({
+    const newEntry = {
```

Comment:
as discussed, please avoid overloading the name `entry` here. Can be called `state` instead.